File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/interfaceInJava/defaultImplementationInInterface/Vehicle.java
package de.fraunhofer.iem.classesAndObjects.interfaceInJava.defaultImplementationInInterface;

/**
 * Demonstrates the default implementation of a method in Interface.
 *
 * @author Ranjith Krishnamurthy
 */
public interface Vehicle {
    String modelNumber = "Its Interface model number";
    String vehicleRegistrationNumber = "Its Interface vehicle registration number";

    public void start();

    public void stop();

    default public void displayModelNumber() {
        System.out.println("Interface implementation = " + modelNumber);
    }

    default public void displayVehicleRegistrationNumber() {
        System.out.println("Interface implementation = " + vehicleRegistrationNumber);
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/interfaceInKotlin/defaultImplementationInInterface/Vehicle.kt
package de.fraunhofer.iem.classesAndObjects.interfaceInKotlin.defaultImplementationInInterface

/**
 * Demonstrates the default implementation of a method in Interface.
 *
 * @author Ranjith Krishnamurthy
 */
interface Vehicle {
    val modelNumber: String
    val vehicleRegistrationNumber: String

    fun start()

    fun stop()

    fun displayModelNumber() {
        println("Interface implementation = $modelNumber")
    }

    fun displayVehicleRegistrationNumber() {
        println("Interface implementation = $vehicleRegistrationNumber")
    }
}